### PR TITLE
[AND-4120] 6.12.0 Max Integration

### PR DESCRIPTION
**Pull request by jiancchen**
- Support RealTime Ads
- Support NativeAds

### CODE REVIEW

File: Vungle/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 6.11.0.2
```
Comment (HuFangshuai) [inline Vungle/CHANGELOG.md]: should be 6.12.0.0?

File: Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java [hunk 2]
```diff
@@ -52,6 +63,73 @@
     // Explicit default constructor declaration
     public VungleMediationAdapter(final AppLovinSdk sdk) { super( sdk ); }
 
+    @Override
+    public void loadNativeAd(MaxAdapterResponseParameters maxAdapterResponseParameters, Activity activity, MaxNativeAdAdapterListener maxNativeAdAdapterListener) {
+
+        //assume as is loaded
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java:75]: typo: assume the ad is loaded

File: Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java [hunk 3]
```diff
@@ -52,6 +63,73 @@
...
+        final Bundle serverParameters = maxAdapterResponseParameters.getServerParameters();
+        final boolean isNativeBanner = serverParameters.getBoolean( "is_native_banner");
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java]: Where do we use the isNativeBanner?

File: Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java [hunk 4]
```diff
@@ -52,6 +63,73 @@
...
+                        .setTitle(nativeAd.getAdTitle())
+                        .setBody(nativeAd.getAdBodyText())
+                        .setMediaView(mediaView)
+                        .setIconView(iconView)
+                        .build();
+
+                List<View> clickableViews = new ArrayList<>();
+                clickableViews.add(maxNativeAd.getIconView());
+                clickableViews.add(maxNativeAd.getMediaView());
+                nativeAd.registerViewForInteraction(nativeAdLayout,
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java]: We should destroy this `nativeAd` instance after the display to trigger reporting.  Assume we could do it in MAX destroy() method.

File: Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java [hunk 5]
```diff
@@ -344,6 +424,8 @@ public void loadAdViewAd(final MaxAdapterResponseParameters parameters, final Ma
     {
         String bidResponse = parameters.getBidResponse();
         boolean isBiddingAd = AppLovinSdkUtils.isValidString( bidResponse );
+        final boolean isNative = parameters.getServerParameters().getBoolean( "is_native" );
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java]: Is this necessary?

File: Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java [hunk 6]
```diff
@@ -52,6 +63,73 @@
...
+        final Bundle serverParameters = maxAdapterResponseParameters.getServerParameters();
+        final boolean isNativeBanner = serverParameters.getBoolean( "is_native_banner");
+        final String placementId = maxAdapterResponseParameters.getThirdPartyAdPlacementId();
+
+        NativeAd nativeAd = new NativeAd(activity, placementId);
+        MediaView mediaView = new MediaView(activity);
+        ImageView iconView = new ImageView(activity);
+        AdConfig adConfig = new AdConfig();
+        NativeAdLayout nativeAdLayout = new NativeAdLayout(activity);
+        nativeAd.loadAd(adConfig, new NativeAdListener() {
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java]: For this 6.12.0 adapter, we should add support for native ad in-app bidding.

File: Vungle/build.gradle.kts [hunk 7]
```diff
@@ -1,62 +1,35 @@
...
-                                appendNode("scope", "compile")
-                            }
-                }
-            }
-        }
-    }
+//    implementation("com.vungle:publisher-sdk-android:6.11.0")
+
+    //TODO: Replace this with maven repo as above when available
+    implementation("com.github.Vungle:vungle-android-sdk:6.12.0-QA2")
```
Comment (HuFangshuai) [inline Vungle/build.gradle.kts]: prefer to replace with the latest version.

File: Vungle/local.properties [hunk 8]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
```
Comment (HuFangshuai) [inline Vungle/local.properties:1]: Is this file not in gitignore?

File: Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java (hunk 6 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java]: Confirmed with iOS, we no need to support native bidding in 6.12.0 max adapter.

File: Vungle/build.gradle.kts (hunk 7 above)
Comment (jiancchen) [reply Vungle/build.gradle.kts]: Done

File: Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java (hunk 4 above)
Comment (jiancchen) [reply Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java]: @HuFangshuai agreed updating

File: Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java (hunk 5 above)
Comment (jiancchen) [reply Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java]: @HuFangshuai actually can be removed

File: Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java [hunk 9]
```diff
@@ -195,7 +222,7 @@ else if ( Vungle.canPlayAd( placementId ) )
             return;
         }
 
-        updateUserPrivacySettings( parameters );
+//        updateUserPrivacySettings( parameters );
```
Comment (felixzhang0703) [inline Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java]: why comment out this line?

**Pull request merged**